librocksdb_sys/crocksdb/c.cc

```diff
@@ -2915,4 +2927,341 @@ const char* crocksdb_pinnableslice_value(const crocksdb_pinnableslice_t* v,
+  crocksdb_table_properties_t props_;
+};
+
+crocksdb_table_properties_collection_t*
+crocksdb_table_properties_collection_create() {
+  return new crocksdb_table_properties_collection_t;
+}
+
+void crocksdb_table_properties_collection_destroy(
+  crocksdb_table_properties_collection_t* collection) {
+  delete collection;
+}
+
+crocksdb_table_properties_collection_iterator_t*
+crocksdb_table_properties_collection_iter_create(crocksdb_table_properties_collection_t* c) {
+  auto it = new crocksdb_table_properties_collection_iterator_t;
+  it->rep_ = c->rep_;
+  it->iter_ = c->rep_->begin();
+  return it;
+}
+
+void crocksdb_table_properties_collection_iter_destroy(
+  crocksdb_table_properties_collection_iterator_t* it) {
+  delete it;
+}
+
+unsigned char crocksdb_table_properties_collection_iter_valid(
+  crocksdb_table_properties_collection_iterator_t* it) {
+  return it->iter_ != it->rep_->end();
+}
+
+void crocksdb_table_properties_collection_iter_next(
+  crocksdb_table_properties_collection_iterator_t* it) {
+  (it->iter_)++;
+}
+
+const char* crocksdb_table_properties_collection_iter_key(
+  crocksdb_table_properties_collection_iterator_t* it, size_t* klen) {
+  if (klen) {
+    *klen = it->iter_->first.size();
+  }
+  return it->iter_->first.data();
+}
+
+crocksdb_table_properties_t* crocksdb_table_properties_collection_iter_value(
+  crocksdb_table_properties_collection_iterator_t* it) {
+  it->props_.rep_ = const_cast<TableProperties*>(it->iter_->second.get());
+  return &it->props_;
+}
+
+/* Table Properties Collector */
+
+struct crocksdb_table_properties_collector_t : public TablePropertiesCollector {
+  void* state_;
+  const char* (*name_)(void*);
+  void (*destruct_)(void*);
+  void (*add_userkey_)(void*,
+                      const char* key, size_t key_len,
+                      const char* value, size_t value_len,
+                      int entry_type, uint64_t seq, uint64_t file_size);
+  void (*finish_)(void*, crocksdb_user_collected_properties_t* props);
+
+  virtual ~crocksdb_table_properties_collector_t() {
+    destruct_(state_);
+  }
+
+  virtual Status AddUserKey(const Slice& key,
+                            const Slice& value,
+                            EntryType entry_type,
+                            SequenceNumber seq,
+                            uint64_t file_size) override {
+    add_userkey_(state_,
+                 key.data(), key.size(),
+                 value.data(), value.size(),
+                 entry_type, seq, file_size);
+    return Status::OK();
+  }
+
+  virtual Status Finish(UserCollectedProperties* rep) override {
+    crocksdb_user_collected_properties_t props;
+    props.rep_ = rep;
+    finish_(state_, &props);
+    return Status::OK();
+  }
+
+  virtual UserCollectedProperties GetReadableProperties() const override {
+    // Seems rocksdb will not return the readable properties and we don't need them too.
+    UserCollectedProperties props;
+    return props;
+  }
```

Comment:
Rocksdb will not return it when we get from tables, not sure if it is a bug, but it is not useful for us anyway.